### PR TITLE
Add support to upload `dnsmasq.conf` and Docker

### PR DESCRIPTION
**Pull request by nferro**
Added support to upload `dnsmasq.conf` and:
* removed Selenium dependency
* added Docker support
* added option to specify admin password as parameter

### CODE REVIEW

**Comment by nferro** [comment]
Hi @Diego-Zulu! 

I took upon your work and changed it a bit to match my necessities while keeping what you did working. The only breaking change is that the custom url for `hosts` is now specified with `-u` instead of `-d` and `-d` will be used to trigger the `hosts` file enablement.

I added Docker support and removed Selenium as I wasn't able to get Selenium working properly.

Let me know if you're interested in merging this to your repo or if I should do a complete fork ;)

File: orbi_dnsmasq/orbi_dnsmasq.py [hunk 1]
```diff
@@ -72,65 +30,80 @@ def telnet_into_router(_user, _pass):
...
-                return driver(webdriver_path)
-            else:
-                return driver()
-        except WebDriverException:
-            continue
-    raise IOError('No selenium web driver was found')
 
+def get_web_ts(host, username, password):
+    r = requests.get(f'http://{host}/debug_detail.htm', auth=(username, password))
+    #print(r.text)
```
Comment (Diego-Zulu) [inline orbi_dnsmasq/orbi_dnsmasq.py]: Let's remove this commented print

File: orbi_dnsmasq/__main__.py [hunk 2]
```diff
@@ -21,14 +29,30 @@
...
+)
+parser.add_option(
+    "-p",
+    "--password",
     action="store",
     default=None,
-    dest="webdriver_path",
-    help="Path to the webdriver for selenium",
+    dest="password",
+    help="The password for the admin user",
```
Comment (Diego-Zulu) [inline orbi_dnsmasq/__main__.py]: Hey, I don't really personally like using this sort of password params, as your password will be recorded in your terminal's history. Nevertheless, I do agree that it's pretty convenient, so as a middle ground I propose changing this description:

```suggestion
    help="The password for the admin user (warning: insecure, watch out for your terminal's history)",
```

What do you think?

File: README.md [hunk 3]
```diff
@@ -70,27 +86,24 @@ Also, **don't worry if you end up running this command twice**. I made sure it d
...
 ```
 
-You can avoid the `-w` flag if you put the downloaded webdriver in your PATH.
+## Known issues
+Sometimes authentication will fail and you'll see a `ts not found, retry in a minute or two` error. Normally retrying works.
 
 ## Possible things to add:
 
-- Read password from ENV var, don't ask it if already found.
+- ~~Read password from ENV var, don't ask it if already found.~~ You can provide the `admin` password with `-p <password>` parameter.
```
Comment (Diego-Zulu) [inline README.md]: Same here, I think we should add:

```suggestion
- ~~Read password from ENV var, don't ask it if already found.~~ You can provide the `admin` password with `-p <password>` parameter (warning: your password may be recorded on your terminal's history, watch out for that)
```

File: .gitignore [hunk 4]
```diff
@@ -108,3 +108,7 @@ venv.bak/
 
 # macOS
 .DS_Store
+
+curls.txt
+*.bak
+tmp/*
```
Comment (Diego-Zulu) [inline .gitignore]: Let's add an extra empty line here

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+RUN pip install requests
+
+ADD . /orbi_dnsmasq
+WORKDIR /orbi_dnsmasq
+RUN python setup.py install
+
+RUN addgroup -S orbi && adduser -S orbi -G orbi
+USER orbi
+
+ENTRYPOINT ["orbi-dnsmasq"]
```
Comment (Diego-Zulu) [inline Dockerfile]: Also one extra empty line here

File: .dockerignore [hunk 6]
```diff
@@ -0,0 +1,5 @@
+__pycache__
+*.pyc
+Dockerfile
+.*
+tmp
```
Comment (Diego-Zulu) [inline .dockerignore]: Also one extra empty line here

**Review by Diego-Zulu — COMMENTED** [review]
Yooo this looks super good! I'm really glad you got to remove selenium and keyboard, those were some giant dependencies that felt like were bloating the project, and not having to worry about outdated selenium drivers is _chef's kiss_

I left some quick comments, after those I would consider this pretty much good to merge!

**Comment by nferro** [comment]
Just committed the changes you requested.

I think the `README.md` needs a big overhaul but currently I don't have time to do it so I just went with your suggestion on saying it's insecure to provide the password as parameter even though I don't agree with it. You can use an environment variable for it that you fill with `read` (or something similar on your shell) so it's not stored in the history.

**Review by Diego-Zulu — APPROVED** [review]

**Comment by Diego-Zulu** [comment]
@all-contributors please add @nferro for development

**Comment by Diego-Zulu** [comment]
@all-contributors please add @nferro for code

**Pull request merged**